projects/plugins/crm/admin/settings/invoicing.page.php

```diff
@@ -33,129 +36,126 @@
 	}
-	$updatedSettings['paymentinfo'] = '';
+	$updated_settings['paymentinfo'] = '';
 	if ( isset( $_POST['paymentinfo'] ) ) {
-		$updatedSettings['paymentinfo'] = zeroBSCRM_textProcess( $_POST['paymentinfo'] );
+		$updated_settings['paymentinfo'] = zeroBSCRM_textProcess( $_POST['paymentinfo'] );
 	}
-	$updatedSettings['paythanks'] = '';
+	$updated_settings['paythanks'] = '';
 	if ( isset( $_POST['paythanks'] ) ) {
-		$updatedSettings['paythanks'] = zeroBSCRM_textProcess( $_POST['paythanks'] );
+		$updated_settings['paythanks'] = zeroBSCRM_textProcess( $_POST['paythanks'] );
 	}
 
 	// } Invoice sending settings
-	$updatedSettings['invfromemail'] = '';
+	$updated_settings['invfromemail'] = '';
 	if ( isset( $_POST['invfromemail'] ) ) {
-		$updatedSettings['invfromemail'] = zeroBSCRM_textProcess( $_POST['invfromemail'] );
+		$updated_settings['invfromemail'] = zeroBSCRM_textProcess( $_POST['invfromemail'] );
 	}
-	$updatedSettings['invfromname'] = '';
+	$updated_settings['invfromname'] = '';
 	if ( isset( $_POST['invfromname'] ) ) {
-		$updatedSettings['invfromname'] = zeroBSCRM_textProcess( $_POST['invfromname'] );
+		$updated_settings['invfromname'] = zeroBSCRM_textProcess( $_POST['invfromname'] );
 	}
 
 	// } Hide Invoice ID
-	$updatedSettings['invid'] = 0;
+	$updated_settings['invid'] = 0;
 	if ( isset( $_POST['wpzbscrm_invid'] ) && ! empty( $_POST['wpzbscrm_invid'] ) ) {
-		$updatedSettings['invid'] = 1;
+		$updated_settings['invid'] = 1;
 	}
 
 	// } Allow Invoice Hash (view and pay without being logged into the portal)
-	// moved to client portal settings 3.0 - $updatedSettings['invhash'] = 0; if (isset($_POST['wpzbscrm_invhash']) && !empty($_POST['wpzbscrm_invhash'])) $updatedSettings['invhash'] = 1;
+	// moved to client portal settings 3.0 - $updated_settings['invhash'] = 0; if (isset($_POST['wpzbscrm_invhash']) && !empty($_POST['wpzbscrm_invhash'])) $updated_settings['invhash'] = 1;
 
 	// } Tax etc
-	$updatedSettings['invtax'] = 0;
+	$updated_settings['invtax'] = 0;
 	if ( isset( $_POST['wpzbscrm_invtax'] ) && ! empty( $_POST['wpzbscrm_invtax'] ) ) {
-		$updatedSettings['invtax'] = 1;
+		$updated_settings['invtax'] = 1;
 	}
-	$updatedSettings['invdis'] = 0;
+	$updated_settings['invdis'] = 0;
 	if ( isset( $_POST['wpzbscrm_invdis'] ) && ! empty( $_POST['wpzbscrm_invdis'] ) ) {
-		$updatedSettings['invdis'] = 1;
+		$updated_settings['invdis'] = 1;
 	}
-	$updatedSettings['invpandp'] = 0;
+	$updated_settings['invpandp'] = 0;
 	if ( isset( $_POST['wpzbscrm_invpandp'] ) && ! empty( $_POST['wpzbscrm_invpandp'] ) ) {
-		$updatedSettings['invpandp'] = 1;
+		$updated_settings['invpandp'] = 1;
+	}
+
+	// } Disable Partial Payments
+	$updated_settings['invoicing_disable_partial_payments'] = 0; // Default to 0 (unchecked)
+	// If the checkbox name exists in POST, it means it was checked.
+	if ( isset( $_POST['wpzbscrm_invoicing_disable_partial_payments'] ) ) {
+		$updated_settings['invoicing_disable_partial_payments'] = 1;
```

Comment:
I'll note that technically this means that a post value of `0` would still result in it being enabled, which is probably why the other settings have the `! empty()` check as well. Not a blocker, but something to consider.